pkg/sensors/exec/exec.go

```diff
@@ -68,6 +71,19 @@ func msgToExecveKubeUnix(m *processapi.MsgExecveEvent, exec_id string, filename
 	return kube
 }
 
+func msgToExecveAccountUnix(m *exec.MsgExecveEventUnix) {
+	mode := cgroups.GetDeploymentMode()
+	if ns, err := namespace.GetMsgNamespaces(m.Unix.Msg.Namespaces); err == nil {
+		if (mode == cgroups.DEPLOY_SD_SERVICE || mode == cgroups.DEPLOY_SD_USER) &&
```

Comment:
Instead of checking `(mode == cgroups.DEPLOY_SD_SERVICE || mode == cgroups.DEPLOY_SD_USER)` can we introduce a user flag for this?

The flag will allow us to document the limitations of just reading `/etc/passwd`. I would call the flag something like: `--resolve-username=unix-basic` and document the limitations.

I would personally have the default be `--resolve-username=disabled`, but I'm fine to set it to "unix-basic" if `(mode == cgroups.DEPLOY_SD_SERVICE || mode == cgroups.DEPLOY_SD_USER)`.

Reply:
But why introduce the flag? what's purpose it is serving? we are already document that in host deployments we are reading /etc/passwd?

Reply:
> But why introduce the flag? what's purpose it is serving? we are already document that in host deployments we are reading /etc/passwd?

I agree with @tixxdz . I don't know is it reasonable to add a flag only for documentation purpose. Maybe we can find a place in docs to describe the restrictions of username resolving? It might help users to be not surprised when for some process event will not have a `username` field. @kkourt what do you think?
P.S. For me it's ok to add on option if you think so.

Reply:
Ok after an offline discussion with @kkourt , indeed it could be problematic on setups that go with specific NSS, even if we ignore non existent entries in /etc/passwd, this falls short on duplicate or even if users want to override entries.

> Instead of checking (mode == cgroups.DEPLOY_SD_SERVICE || mode == cgroups.DEPLOY_SD_USER) can we introduce a user flag for this?

The deployment is a guard, cause if in container deployment or something else, it does not make sense to check that container passwd file.

> The flag will allow us to document the limitations of just reading /etc/passwd. I would call the flag something like: --resolve-username=unix-basic and document the limitations.

Makes sense, what if we make it: `--metadata-username=unix` ? to reflect on  https://linux.die.net/man/8/pam_unix and  metadata enrichment?   calling it --resolve-username is fine too.

> I would personally have the default be --resolve-username=disabled, but I'm fine to set it to "unix-basic" if (mode == cgroups.DEPLOY_SD_SERVICE || mode == cgroups.DEPLOY_SD_USER).

Yes that works, let's go with disabled meaning just that, we check deployment mode and we set it, and if in future we want to disable it even for host deployments we could do that.

BTW some notes as follow up:
* Cache the entries. Taking the sssd architecture https://sssd.io/docs/architecture.html it has a cache by default, I think entries could expire, but for entries below 1000 , and referencing systemd uids https://github.com/systemd/systemd/blob/main/docs/UIDS-GIDS.md#summary  we can cache those in small map.
* If uid 0 resolves to root which is the convention we should not query the file again, but for later.

Reply:
The primary motivation for having a flag is:
* allow users to turn off this feature (this is aimed at environments that rely on other methods. e.g., NSS
* allow us to support other methods in the future

I prefer `--username-metadata=unix` to `--metdata-username`.

One of the extensions we might want to do is `--username-metdata=unix:<file>` and then mount `/etc/passwd` inside the tetragon's pod to location `<file>`. Doing so will allow us to get host username events even if we deploy via a K8s daemonset.

Reply:
Yes `--username-metadata=unix ` is good too. 

> One of the extensions we might want to do is --username-metdata=unix:<file> and then mount /etc/passwd inside the tetragon's pod to location <file>. Doing so will allow us to get host username events even if we deploy via a K8s daemonset.

Hmm I dont think we should do that, security scanners may flag tetragon. Would be better to ask users that want this, install  [systemd-userdb](https://manpages.ubuntu.com/manpages/lunar/man8/systemd-userdbd.8.html) then mount the socket  from `/run/systemd/userdb/ `  then do json queries on the records.  Also container dynamic userids IIRC are not registered somewhere or propagated so we can look them up IIRC.

Reply:
Guys, I added `--username-metadata` option. I also added a cache for username lookups by UID below 1000.
I looked at this [table](https://github.com/systemd/systemd/blob/main/docs/UIDS-GIDS.md#summary) and I'm not sure that systemd UIDs greater than 1000 are not changing. Maybe cache only UIDs from [0; 1000) ?

Reply:
> Guys, I added --username-metadata option

Thanks @anfedotoff.
I'll let @tixxdz answer about the cache, but in my opinion we can merge the PR without it.

Reply:
> > Guys, I added --username-metadata option
> 
> Thanks @anfedotoff. I'll let @tixxdz answer about the cache, but in my opinion we can merge the PR without it.

I agree, let's first merge what we have now, and you can add the optimization to not open every time /etc/passwd in follow up PR